src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs

```diff
@@ -91,30 +94,69 @@ public Task Stop()
         /// <returns>A <see cref="TransactionBriefModel"/> or <see cref="TransactionVerboseModel"/> as specified by verbose. <c>null</c> if no transaction matching the hash.</returns>
         /// <exception cref="ArgumentException">Thrown if txid is invalid uint256.</exception>"
-        /// <remarks>Requires txindex=1, otherwise only txes that spend or create UTXOs for a wallet can be returned.</remarks>
+        /// <remarks>When called with a blockhash argument, getrawtransaction will return the transaction if the specified block is available and the transaction is found in that block.
+        /// When called without a blockhash argument, getrawtransaction will return the transaction if it is in the mempool, or if -txindex is enabled and the transaction is in a block in the blockchain.</remarks>
         [ActionName("getrawtransaction")]
         [ActionDescription("Gets a raw, possibly pooled, transaction from the full node.")]
-        public async Task<TransactionModel> GetRawTransactionAsync(string txid, int verbose = 0)
+        public async Task<TransactionModel> GetRawTransactionAsync(string txid, int verbose = 0, string blockHash = null)
         {
-            uint256 trxid;
-            if (!uint256.TryParse(txid, out trxid))
-                throw new ArgumentException(nameof(txid));
+            Guard.NotEmpty(txid, nameof(txid));
 
-            Transaction trx = this.pooledTransaction != null ? await this.pooledTransaction.GetTransaction(trxid) : null;
+            if (!uint256.TryParse(txid, out uint256 trxid))
+            {
+                throw new ArgumentException(nameof(trxid));
+            }
 
-            if (trx == null)
+            uint256 hash = null;
+            if (!string.IsNullOrEmpty(blockHash) && !uint256.TryParse(blockHash, out hash))
             {
-                trx = this.blockStore != null ? await this.blockStore.GetTransactionByIdAsync(trxid) : null;
+                throw new ArgumentException(nameof(blockHash));
             }
 
-            if (trx == null)
-                return null;
+            // Special exception for the genesis block coinbase transaction.
+            if (trxid == this.Network.GetGenesis().GetMerkleRoot().Hash)
+            {
+                throw new RPCServerException(RPCErrorCode.RPC_INVALID_ADDRESS_OR_KEY, "The genesis block coinbase is not considered an ordinary transaction and cannot be retrieved.");
+            }
+
+            Transaction trx = null;
+            ChainedHeaderBlock chainedHeaderBlock = null;
 
+            if (hash == null)
+            {
+                // Look for the transaction in the mempool, and if not found, look in the indexed transactions.
+                trx = (this.pooledTransaction == null ? null : await this.pooledTransaction.GetTransaction(trxid)) ??
+                      await this.blockStore.GetTransactionByIdAsync(trxid).ConfigureAwait(false);
```

Comment:
here probbaly also use `ConsensusManager.GetBlockDataAsync`

Reply:
to get a block? I need to get a transaction.